### PR TITLE
Basic Obstacle Avoidance is modified (!!!This was initial pull request for the assignment, but used another branch later on)

### PR DESCRIPTION
**Pull request by ediz-genc**
##########################################################
**Basic Obstacle Avoidance**
##########################################################
**type**
# Assignment 
**git tags**
# Assignment
**changes**
# Solution for basic obstacle avodiance
**notes**
---

### CODE REVIEW

**Review by bassamEldesouki — APPROVED** [review]

**Review by bassamEldesouki — APPROVED** [review]

File: arduino/smartcar/smartcar.ino [hunk 1]
```diff
@@ -49,7 +58,12 @@ void setup()
     Serial.begin(9600);
 }
 
+// Distance of our choice is 25cm from the wall.
 void loop()
 {
-  
-}
+    car.setSpeed(70);
+    while(!(frontUS.getDistance() == 0) && frontUS.getDistance() < 90 ){
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino:65]: There are two issues:
1. Why a loop (i.e. `while`) inside another loop (i.e. `loop()`)? If you want something to only happen once like setting the speed, you place it in `setup()`.
2. You are calling `getDistance` **twice** which means you will get two **different** measurements. Is this really what you want? What happens if the first time the sensor returns 100 and the next time it returns 0?

**Review by platisd — CHANGES_REQUESTED** [review]

**Review by ediz-genc — COMMENTED** [review]
I believe my colleague has pushed the assignment with the necessary changes!.

File: arduino/smartcar/smartcar.ino (hunk 1 above)
Comment (ediz-genc) [reply arduino/smartcar/smartcar.ino:65]: I believe my colleague has pushed the assignment with the necessary changes!.

**Comment by platisd** [comment]
> I believe my colleague has pushed the assignment with the necessary changes!.

I don't know what you are referring to but there are no new changes. Not in this pull request at least.

**Comment by platisd** [comment]
Also avoid creating a fork of the repo and then creating a pull request from there. You all have access to create your own branches here.
If you really want to work on forked repos, then please make CI run on `pull_request` too.

**Comment by ediz-genc** [comment]
> Also avoid creating a fork of the repo and then creating a pull request from there. You all have access to create your own branches here.
> If you really want to work on forked repos, then please make CI run on `pull_request` too.

OK, will do thanks. Could you help me to resolve this merge request? (we don't need this pull request though since colleague of mine used another branch for the assignment!)

**Comment by platisd** [comment]
> > Also avoid creating a fork of the repo and then creating a pull request from there. You all have access to create your own branches here.
> > If you really want to work on forked repos, then please make CI run on `pull_request` too.
> 
> OK, will do thanks. Could you help me to resolve this merge request? (we don't need this pull request though since colleague of mine used another branch for the assignment!)

If you don't need it I guess you can close it? Or did I misunderstand something?

**Comment by ediz-genc** [comment]
> > > Also avoid creating a fork of the repo and then creating a pull request from there. You all have access to create your own branches here.
> > > If you really want to work on forked repos, then please make CI run on `pull_request` too.
> > 
> > 
> > OK, will do thanks. Could you help me to resolve this merge request? (we don't need this pull request though since colleague of mine used another branch for the assignment!)
> 
> If you don't need it I guess you can close it? Or did I misunderstand something?

This was the initial assignment pull request, but a colleague of mine used today another bench to complete the assignment. I figured this pull request is not needed anymore, am I right? If it is not needed, then the best is to close it or something else?

**Comment by platisd** [comment]
Yeap, it should be closed 👍

**Comment by ediz-genc** [comment]
!!!This was the initial pull request for the assignment but used another branch later on.

**Pull request closed without merging**